jib-core/src/test/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStepTest.java

```diff
@@ -127,4 +138,80 @@ public void testCall_offlineMode_cached()
 
     Mockito.verify(buildContext, Mockito.never()).newBaseImageRegistryClientFactory();
   }
+
+  @Test
+  public void testObtainPlatformSpecificImageManifest()
+      throws IOException, InterruptedException, GeneralSecurityException, URISyntaxException,
+          RegistryException {
+    String testManifestListTemplate =
+        " {\n"
+            + "   \"schemaVersion\": 2,\n"
+            + "   \"mediaType\": \"application/vnd.docker.distribution.manifest.list.v2+json\",\n"
+            + "   \"manifests\": [\n"
+            + "      {\n"
+            + "         \"mediaType\": \"application/vnd.docker.distribution.manifest.v2+json\",\n"
+            + "         \"size\": 424,\n"
+            + "         \"digest\": \"sha256:f67dcc5fc786f04f0743abfe0ee5dae9bd8caf8efa6c8144f7f2a43889dc513b\",\n"
+            + "         \"platform\": {\n"
+            + "            \"architecture\": \"testArchitecture\",\n"
+            + "            \"os\": \"testOS\"\n"
+            + "         }\n"
+            + "      },\n"
+            + "      {\n"
+            + "         \"mediaType\": \"application/vnd.docker.distribution.manifest.v2+json\",\n"
+            + "         \"size\": 425,\n"
+            + "         \"digest\": \"sha256:5bb8e50aa2edd408bdf3ddf61efb7338ff34a07b762992c9432f1c02fc0e5e62\",\n"
+            + "         \"platform\": {\n"
+            + "            \"architecture\": \"Architecture\",\n"
+            + "            \"os\": \"OS\"\n"
+            + "         }\n"
+            + "      }\n"
+            + "   ]\n"
+            + "}";
+
+    String testManifestTemplate =
+        "{\n"
+            + "    \"schemaVersion\": 2,\n"
+            + "    \"mediaType\": \"application/vnd.docker.distribution.manifest.v2+json\",\n"
+            + "    \"config\": {\n"
+            + "        \"mediaType\": \"application/vnd.docker.container.image.v1+json\",\n"
+            + "        \"size\": 424,\n"
+            + "        \"digest\": \"sha256:5bb8e50aa2edd408bdf3ddf61efb7338ff34a07b762992c9432f1c02fc0e5e62\"\n"
+            + "    }\n"
+            + "}";
+    Mockito.when(buildContext.getContainerConfiguration()).thenReturn(containerConfiguration);
+    Mockito.when(containerConfiguration.getPlatforms())
+        .thenReturn(ImmutableSet.of(new Platform("testArchitecture", "testOS")));
```

Comment:
I think it's more meaningful to make the code return the second manifest from the list.
```suggestion
        .thenReturn(ImmutableSet.of(new Platform("targetArchitecture", "targetOS")));
```